examples/gimli/src/gimli.rs

```diff
@@ -75,7 +75,13 @@ fn squeeze_block(s: State) -> Block {
 
 fn gimli_hash_state(input: &ByteSeq, mut s: State) -> State {
     let rate = Block::length();
-    for i in 0..input.num_chunks(rate) {
+    let mut chunks = input.num_chunks(rate);
+
+    if input.len() % 16 == 0 {
+        chunks += 1;
+    }
+
+    for i in 0..chunks {
```

Comment:
Ok so this works as a fix because `get_chunk` will simply return an empty block when there's nothing left. But it's a little odd to knowingly run over the end of the input.
I would suggest to instead move the `else` part out of the for loop and always do it in the end for the remaining, incomplete block, or the empty block if there's nothing left.

You could maybe also use `num_exact_chunks` and `get_exact_chunk` here to make this cleaner (these two functions got introduced after this code was written). You can look at chacha20.rs for an example of this approach.

Reply:
Yes sure! That is also how it has been implemented in the python version I believe.